src/main.rs

```diff
@@ -103,6 +112,7 @@ pub fn start<B>(
     terminal_backend: B,
```

Comment:
Currently, the `start` function takes a growing list of parameters for various options and flags (e.g., `show_apparent_size`, `disable_delete_confirmation`). As we continue to add more options, this signature risks becoming unwieldy and harder to maintain. The challenge became especially apparent when updating the UI tests, which required significant changes due to the expanded parameter list.

To address this, I’ve started a WIP implementation that introduces an `Opt` struct as the last argument in `start`. This encapsulates all optional parameters into a single structure, aiming to reduce churn when adding or modifying options.

I’d appreciate input on the following:
- Is encapsulating options in a configuration struct like `Opt` an idiomatic approach in Rust?
- Would a builder pattern provide better flexibility and maintainability for this use case?
- Are there other patterns or techniques in Rust that would align well with the goal of minimizing signature changes and improving testability?

I’m especially interested in feedback on how this impacts code readability, maintainability, and idiomatic usage in Rust.